packages/medusa/src/services/cart.ts

```diff
@@ -2604,6 +2604,7 @@ class CartService extends TransactionBaseService {
       .withTransaction(transactionManager)
       .delete({
         item_id: nonReturnLineIDs,
+        discount_id: Not(IsNull()),
```

Comment:
question: Is it possible to have a custom adjustment linked to a discount?

Reply:
question: Should we have the same behavior on order-edit?

Reply:
Good point, the ticket refers to `refreshAdjustments_` specifically but I think we should add this on OE too because we can end up with deleted custom adjustments via OE as well.
And regarding your first question, let's see what @olivermrbl and @srindom will say.

Reply:
> Is it possible to have a custom adjustment linked to a discount?

Imo. adjustments are not custom if they are linked to a discount. That would be contradictory to their purpose.

Reply:
Make sense 💪